apm-agent-plugins/apm-jaxrs-plugin/src/main/java/co/elastic/apm/agent/jaxrs/JaxRsTransactionHelper.java

```diff
@@ -0,0 +1,57 @@
+package co.elastic.apm.agent.jaxrs;
+
+import co.elastic.apm.agent.bci.VisibleForAdvice;
+import co.elastic.apm.agent.impl.ElasticApmTracer;
+import co.elastic.apm.agent.impl.transaction.Transaction;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.lang.annotation.Annotation;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+
+@VisibleForAdvice
+public class JaxRsTransactionHelper {
+
+    private final JaxRsConfiguration jaxRsConfiguration;
+
+    @VisibleForAdvice
+    public JaxRsTransactionHelper(ElasticApmTracer tracer) {
+        this.jaxRsConfiguration = tracer.getConfig(JaxRsConfiguration.class);
+    }
+
+    @VisibleForAdvice
+    public void setTransactionName(@Nonnull Transaction currentTransaction,
```

Comment:
That's gonna be too much runtime overhead. We could do it similar to `co.elastic.apm.agent.bci.bytebuddy.SimpleMethodSignatureOffsetMappingFactory` which only computes it once and stores it as a string constant.